### ISSUE TITLE
STA-9: Anchor project setup (Anchor.toml, Cargo.toml, program scaffold)

### ISSUE THREAD

**Issue by Puneethkumarck** [Puneethkumarck/stablepay-hackathon#9]
## Story: Anchor Project Setup

**Parent:** #1 (Anchor Escrow Program)
**Dependencies:** None
**Standards:** Read [docs/SOLANA_CODING_STANDARDS.md](docs/SOLANA_CODING_STANDARDS.md) before starting.

## Context

Scaffold the Anchor project following the modular file layout from SOLANA_CODING_STANDARDS.md §1. This creates the skeleton that all subsequent instructions (#10–#14) build on. The program will hold USDC in escrow PDAs for cross-border remittances on Solana devnet.

## Acceptance Criteria

- [ ] `Anchor.toml` configured for devnet with program keypair and `[programs.devnet]` entry
- [ ] `programs/stablepay-escrow/Cargo.toml` with `anchor-lang = "0.30"`, `anchor-spl = "0.30"`, `overflow-checks = true` in `[profile.release]`, and clippy lints (`cast_possible_truncation = "deny"`, `cast_sign_loss = "deny"`)
- [ ] Modular file layout created per SOLANA_CODING_STANDARDS.md §1:
  ```
  programs/stablepay-escrow/src/
    lib.rs                    # declare_id!, mod declarations, #[program] thin dispatcher
    instructions/
      mod.rs                  # pub mod for each instruction (empty stubs for now)
    state/
      mod.rs                  # pub mod escrow (empty stub)
    errors.rs                 # Empty #[error_code] enum placeholder
    constants.rs              # ESCROW_SEED, VAULT_SEED, USDC devnet mint address
  ```
- [ ] `lib.rs` follows the thin dispatcher pattern — only `declare_id!`, module declarations, and `#[program]` delegation
- [ ] `constants.rs` defines `ESCROW_SEED`, `VAULT_SEED`, and USDC devnet mint pubkey
- [ ] `anchor build` compiles successfully
- [ ] `cargo fmt -- --check` passes
- [ ] `cargo clippy -- -D warnings` passes

## Agent Notes

- Use `declare_id!("YourProgramId...")` with the generated keypair from `anchor keys list`
- Do NOT put business logic in `lib.rs` — it is a thin dispatcher only
- No wildcard imports from external crates except `anchor_lang::prelude::*`
- All seed constants must be `&[u8]` (e.g., `pub const ESCROW_SEED: &[u8] = b"escrow";`)
